src/classes/ERR_Handler.cls

```diff
@@ -71,137 +71,236 @@ public class ERR_Handler {
 	                errors.errorRecords.add(err);
-	                
-	                //We cannot use addError on the record here, because if we do, the whole transaction will be rolled back, and 
-	                //no error record will be saved,or error notification sent.
-	                
+
+	                //We cannot use addError on the record here, because if we do, the whole transaction will be rolled back, and
+	                //no error record will be saved, or error notification sent.
+
 	                //Displaying an error message to the user
 	                if(ApexPages.currentPage() != null)
-	                   ApexPages.addMessage(new ApexPages.Message(ApexPages.Severity.ERROR, err.Full_Message__c)); 	                
+	                   ApexPages.addMessage(new ApexPages.Message(ApexPages.Severity.ERROR, err.Full_Message__c));
 	        	}
 	        }
     	}
     	return errors;
     }
-        
+
+	/*******************************************************************************************************
+    * @description Processes the results of a group DML operations caused by a batch, future, or queued job and performed
+    * 				using the Database class to determine if errors occurred. It does not store errors or send error email
+    * 				notifications. These operations should be performed after the rollback.
+    *
+    * @param dmlResults The results of a DML batch operation performed using the Database class.
+    * @param sObjects The records the DML operation was performed on.
+    * @return Errors The errors with a single list that occurred during a batch DML operation performed using the Database class.
+    */
+	public static Errors getJobErrors(List<Object> dmlResults, List<SObject> sObjects) {
+		Errors errors = new Errors();
+		List<String> errorsFound = new List<String>();
+
+		if (dmlResults != null) {
+			for (Integer i=0; i<dmlResults.size(); i++) {
+
+				Boolean isSuccess = isDmlSuccess(dmlResults[i]);
+
+				if (!isSuccess) {
+					List<Database.Error> errList = getDmlResultErrors(dmlResults[i]);
+					Integer totalErrors = 0;
+					String errDetailsMessage = '';
+
+					for (Database.Error err : errList) {
+						totalErrors++;
+						errDetailsMessage += err.getStatusCode() + ': ' + ERR_ExceptionHandler.beautifyExceptionMessage(err.getMessage());
+
+						if (errDetailsMessage.endsWith('.')){
+							errDetailsMessage += ' ';
+						} else if (!errDetailsMessage.endsWith('. ')){
+							errDetailsMessage += '. ';
+						}
+					}
+
+					String fullObjectErrMessage = 'There were ' + totalErrors + ' errors saving to object ' + sObjects[i].Id + '. ' + errDetailsMessage;
+
+					errorsFound.add(fullObjectErrMessage);
+				}
+			}
+			if (errorsFound.size() > 0){
+				String errorMessage = String.join(errorsFound, '\n* ');
+				errors.errorsExist = true;
+
+                String objectType = sObjects[1].getSObjectType().getDescribe().getName();
```

Comment:
There are probably alternative approaches to get the ObjectType as well as match types in the `UTIL_Describe` class. My biggest concern is using getDescibe() in a for loop in the `areObjectsSameType()` method below. Plus, you could technically compare `SObjectType` instead of comparing against the object name.

Reply:
I see there's method isObjectIdThisType in UTIL_Describe, but it also requires a full getDescribe for each object. I'll trim it down to just call getSObjectType(), which should hopefully improve performance, but let me know if you think it would be better to use isObjectIdThisType. I'll also make sure I correct to using spaces :)